GoogleForCreators/web-stories-wp
# Enable custom page templates feature

## Context
Enables the custom page templates feature


## Summary
See [Figma](https://www.figma.com/file/1YM3jA9h4QGc4xLdyaXIJS/Stories-Sprint-1.7?node-id=7524%3A9254)


## Relevant Technical Choices
Currently, only the first 100 page templates are fetched by default. We should define what this number should be and if we can[ fetch all the templates](https://github.com/google/web-stories-wp/pull/7268#discussion_r618416479) or should implement pagination for the virtual list.


## To-do

- [x] Enable feature
- [x] Add Karma tests (still missing some parts to be able to add them all)



## User-facing changes



## Testing Instructions



### QA






This PR can be tested by following these steps:

1. Open the Page templates panel in the library
2. Verify the dropdown for choosing between the Default and Saved templates is present
3. Click on the "Save current page as template" button.
4. Verify that "Saved templates" section will be automatically opened and the new template is temporarily shown blue highlight around it.
5. Verify the "Save current page as template" button looks like [in Figma](https://www.figma.com/file/1YM3jA9h4QGc4xLdyaXIJS/Stories-Sprint-1.7?node-id=7524%3A9254).
6. Verify again generally that all the main actions work as expected, including removing a saved template and applying a template (saved or default). Applying a template should add it as a new Page without confirmation, removing a template should ask for confirmation.

### UAT







This PR can be tested by following these steps:

1.
## Checklist




---



Fixes #7065
